### PR TITLE
Black Formatting

### PR DESCRIPTION
**Pull request by ssolson**
This PR adds starts to address formatting issues in MHKiT by adding black:
- Black is an auto-formatter, not a Python linter or interpreter. 
- https://github.com/psf/black

This PR adds black because it will automatically format our files allowing our users to focus on coding and letting black fix the formatting issues automatically.

To adhere to black formatting developers can install and run black from the terminal manually, they can install black as an extension in their IDE such as VS code, or they can enable enable the pre-commit hook.

What this PR changes:
- Ran  `black .`  formatting on the repo
  - After installing black I auto reformatted the repository using the above command
- add an GitHub actions check that code adheres to black
- Added an optional pre commit that will auto format files to adhere to black on commits should a developer choose to use it. 
- Added test coverage so that tests would pass

### CODE REVIEW

**Comment by ssolson** [comment]
@akeeste this is ready for review

File: mhkit/tests/wave/test_contours.py [hunk 1]
```diff
@@ -1,290 +1,585 @@
...
+    #             log_kde_copula["bivariate_KDE_log_x1"],
+    #             self.wdrt_copulas["bivariate_KDE_log_x1"],
+    #         ),
+    #         np.allclose(
+    #             log_kde_copula["bivariate_KDE_log_x2"],
+    #             self.wdrt_copulas["bivariate_KDE_log_x2"],
+    #         ),
+    #     ]
+    #     self.assertTrue(all(close))
+
```
Comment (akeeste) [inline mhkit/tests/wave/test_contours.py:544]: should these tests still be commented? Let's remove or add back in

**Comment by akeeste** [comment]
Noting here some non-black related changes, @ssolson add anything else I've missed that you did. We've largely discussed all of these and they aren't issues, but noting for future reference:
- new API key added for NREL's HSDS server
- ``mhkit.tests.river.test_io`` broken out into ``mhkit.tests.river.test_io_d3d`` and ``mhkit.tests.river.test_io_usgs``
- added functional tests and error handing tests to ``mhkit.tests.dolfyn.test_tools, mhkit.tests.mooring.test_mooring``
- added error handling tests for ``mhkit.tests.river.test_resource, mhkit.tests.wave.io.hindcast.test_wind_toolkit, mhkit.tests.wave.test_contours, mhkit.tests.loads.test_loads``

File: mhkit/tests/wave/test_contours.py (hunk 1 above)
Comment (ssolson) [reply mhkit/tests/wave/test_contours.py:544]: Good Catch. These should be added back. They are a significant source of time during testing.

**Comment by ssolson** [comment]
> Noting here some non-black related changes, @ssolson add anything else I've missed that you did. We've largely discussed all of these and they aren't issues, but noting for future reference:
> 
> * new API key added for NREL's HSDS server
> * `mhkit.tests.river.test_io` broken out into `mhkit.tests.river.test_io_d3d` and `mhkit.tests.river.test_io_usgs`
> * added functional tests and error handing tests to `mhkit.tests.dolfyn.test_tools, mhkit.tests.mooring.test_mooring`
> * added error handling tests for `mhkit.tests.river.test_resource, mhkit.tests.wave.io.hindcast.test_wind_toolkit, mhkit.tests.wave.test_contours, mhkit.tests.loads.test_loads`

Thank you for documenting all these.

**Comment by ssolson** [comment]
> * with my current MHKiT environment I had some dependency conflicts (jupyter-server and pywinpty) when installing pre-commit. I tried upgrading a those specific packages which broke others (jupyterlab, jupyterlab-server). I have not tried reinstalling everything at once yet. Did you get pre-commit installed okay with the rest of your MHKiT environment?

Adam as I mentioned in the meeting I uninstalled the pre-commit as I found it slow.

I just reinstalled it and I did not get any conflicts.
image

**Comment by akeeste** [comment]
> > * with my current MHKiT environment I had some dependency conflicts (jupyter-server and pywinpty) when installing pre-commit. I tried upgrading a those specific packages which broke others (jupyterlab, jupyterlab-server). I have not tried reinstalling everything at once yet. Did you get pre-commit installed okay with the rest of your MHKiT environment?
> 
> Adam as I mentioned in the meeting I uninstalled the pre-commit as I found it slow.
> 
> I just reinstalled it and I did not get any conflicts. image

That sounds good, thanks for checking. I just wanted to make sure that it *could* work, even if its slow since we call it out as an option in the readme.

**Comment by akeeste** [comment]
Once the commented tests are back in this should be good to go! Thanks @ssolson

**Pull request merged**